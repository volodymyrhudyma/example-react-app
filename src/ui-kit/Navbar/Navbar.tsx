import React from 'react';
import { Wrapper, LogoWrapper, ItemsWrapper } from './styled/Navbar.styled';
import { IMenuItem } from 'src/types';
import NavbarItem from '../NavbarItem/NavbarItem';
import Icon from '../Icon/Icon';
import icons from 'src/constants/icons';

interface INavbarProps {
  items: IMenuItem[];
}

const Navbar = ({ items }: INavbarProps) => (
  <Wrapper>
    <LogoWrapper href="/">
      <Icon src={icons.logo} height={42} width={30} />
    </LogoWrapper>
    <ItemsWrapper>
      {items.map((item, index) => (
        <NavbarItem key={item.icon} item={item} isActive={index === 0} />
      ))}
    </ItemsWrapper>
  </Wrapper>
);

export default Navbar;
