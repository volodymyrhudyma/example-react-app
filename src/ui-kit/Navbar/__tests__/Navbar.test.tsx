import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../Navbar';

describe('Navbar component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Navbar
        items={[
          {
            icon: 'test',
            link: '/test',
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
