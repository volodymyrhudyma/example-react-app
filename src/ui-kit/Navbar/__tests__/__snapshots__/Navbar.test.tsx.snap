// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
<styled.div>
  <styled.a
    href="/"
  >
    <Icon
      alt="Icon"
      height={42}
      src="/icons/logo.png"
      width={30}
    />
  </styled.a>
  <styled.div>
    <NavbarItem
      isActive={true}
      item={
        Object {
          "icon": "test",
          "link": "/test",
        }
      }
      key="test"
    />
  </styled.div>
</styled.div>
`;
